File: backend/schemas/patient.py
from datetime import datetime

from pydantic import BaseModel, EmailStr, ConfigDict


class PatientCreate(BaseModel):
    medical_record_number: str
    first_name: str
    last_name: str
    date_of_birth: datetime | None = None
    gender: str | None = None
    phone_number: str | None = None
    email: EmailStr | None = None
    communication_consent: bool = True
    preferred_language: str = "English"
    preferred_call_time: str | None = None


class PatientResponse(PatientCreate):
    id: int
    hospital_id: int
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/patient.ts
export interface Patient {
  id: number
  hospital_id: number
  mrn: string
  medical_record_number?: string
  first_name: string
  last_name: string
  date_of_birth?: string | null
  gender?: string | null
  phone?: string | null
  phone_number?: string | null
  email?: string | null
  communication_consent: boolean
  preferred_language?: string | null
  preferred_call_time?: string | null
  is_active: boolean
  created_at?: string
  updated_at?: string
}
